### DISCUSSION TITLE
Username & Password Authentication

### DISCUSSION THREAD

**Discussion by ghost** [hufrea/byedpi#41] [Ideas]
Привет!
Спасибо за такую прекрасную вещь как ByeDPI !
Планируется ли добавление возможности аутентификации на сервере по логину/паролю и доступа к серверу авторизованным пользователям или поддержка linux pam?

**Comment by hufrea** [comment]
Здравствуйте.
Не планируется, т.к. прокси задуман как локальный, т.е. работающий либо на самой машине, либе в доверенной сети (локалке).

**Reply by themaximsuper** [reply]
Здравствуйте. Т.е. никак по логину и паролю подключиться нельзя? Просто хочу настроить tun2socks через ваш прокси. Tun2socks предполагает авторизацию в socks4 по только id и socks5 только по логину-паролю.

**Reply by hufrea** [reply]
> Т.е. никак по логину и паролю подключиться нельзя?

Нет. В tun2socks авторизация не объязательна.

**Reply by themaximsuper** [reply]
[Вот тут](https://github.com/xjasonlyu/tun2socks/wiki/Proxy-Models) указано, что для использования socks4-прокси нужен USERID, а для socks5 - имя пользователя и пароль. При подключении без этих данных пишется ошибка, как я понял, в виду неподдержки авторизации. Я пытался подключаться и по HTTP, но, в этом случае, у меня нет возможности передачи UDP запросов.

**Reply by ProMix0** [reply]
> [Вот тут](https://github.com/xjasonlyu/tun2socks/wiki/Proxy-Models) указано

Вот тут указано, что SOCKS4 поддерживает аутентификацию только по USERID. Это не означает, что аутентификация обязательна

Что вы пытаетесь сделать, что, как и через что настроить?

**Reply by themaximsuper** [reply]
> Вот тут указано, что SOCKS4 поддерживает аутентификацию только по USERID. Это не означает, что аутентификация обязательна

Да, я понимаю, я и попытался без аутентификации всё это сделать, но выдаётся ошибка.

> Что вы пытаетесь сделать, что, как и через что настроить?

Я пытаюсь настроить виртуальный шлюз с помощью tun2socks, весь трафик с которого хочу перенаправлять через byedpi.

**Reply by themaximsuper** [reply]
Судя по ошибкам, у меня идут UDP запросы то ли на DNS, то ли с него, и они не проходят через обычный HTTP-прокси (я пробовал PowerTunnel), поэтому решил попробовать через byedpi, так как заявлено, что он использует протокол SOCKS, который может подразумевать проксирование UDP-запросов.
